jax/lax.py

```diff
@@ -416,6 +416,35 @@ def sort_key_val(keys, values, dimension=-1):
   return sorted_keys, sorted_values
 
 def while_loop(cond_fun, body_fun, init_val):
+  """Call `body_fun` repeatedly in a loop while `cond_fun` is True.
+
+  Arguments:
+    cond_fun: pure function of type `T -> Bool`.
+    body_fun: pure function of type `T -> T`.
+    init_val: value of type `T`, a type that can be a scalar, array, or any
+      (nested) Python tuple/list/dict thereof.
+
+  Returns:
+    The output from the final iteration of body_fun, of type `T`.
+
+  The semantics of `while_loop` are given by this Python implementation:
+
+    def while_loop(cond_fun, body_fun, init_val):
+      val = init_val
+      while cond_fun(val):
+        val = body_fun(val)
+      return val
+
+  Unlike that pure Python version, `while_loop` is a JAX primitive and is
+  lowered to a single XLA While HLO. That makes it useful for reducing
+  compilation times for jit-compiled functions, since native Python loop
+  constructs in an `@jit` function are unrolled, leading to large XLA
+  computations.
+
+  Another difference from using Python-native loop constructs is that
+  `while_loop` is not reverse-mode differentiable because XLA computations
```

Comment:
Perhaps `... not (yet) reverse-mode differentiable ...`

Reply:
Done.